### PR TITLE
Add redis pool configuration for metricbeat

### PR DESCRIPTION
**Pull request by chrsblck**
- Extend MetricSeter interface with a Setup function
- Add hook to Setup/Configure MetricSeters

### CODE REVIEW

File: metricbeat/helper/metricset.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ type MetricSet struct {
 // Creates a new MetricSet
 func NewMetricSet(name string, new func() MetricSeter, module *Module) *MetricSet {
 	metricSeter := new()
+	metricSeter.Setup(module.cfg)
```
Comment (ruflin) [inline metricbeat/helper/metricset.go]: We should probably check for an error here. The consequence is, that NewMetricSet should also be able to return an error. I can add that later for you.

File: metricbeat/metricbeat.yml [hunk 2]
```diff
@@ -9,6 +9,8 @@ metricbeat:
       metricsets: ["info"]
       period: 1s
       hosts: ["127.0.0.1:6379"]
+      network: tcp
```
Comment (ruflin) [inline metricbeat/metricbeat.yml]: Config changes should always be introduced in beat.yml and then by running `make update` the metricbeat.yml is updated.

File: metricbeat/module/redis/info/info.go [hunk 3]
```diff
@@ -129,21 +129,51 @@ func init() {
...
+		m.redisPools[host] = redisPool
+	}
+	return nil
+}
+
+func (m *MetricSeter) Fetch(ms *helper.MetricSet) (events []common.MapStr, err error) {
+	for _, host := range ms.Config.Hosts {
+		c := m.redisPools[host].Get()
+		out, err := rd.String(c.Do("INFO"))
+		c.Close()
```
Comment (ruflin) [inline metricbeat/module/redis/info/info.go]: Do we need to close it after every Fetch call?

**Comment by ruflin** [comment]
The change look goods to me. Run `make fmt` to get the build green (cleanup coding style).

Is it doing what you expected it to do?

**Comment by ruflin** [comment]
jenkins, test it

File: metricbeat/module/redis/info/info.go (hunk 3 above)
Comment (chrsblck) [inline metricbeat/module/redis/info/info.go]: Yes, this releases the connection back to the pool.

**Comment by chrsblck** [comment]
@ruflin It's working as I expected. Though there was a failing test that I missed, should be fixed now.

The `fmt`ing issue was me having `[] Format Code` checked in Intellij on commit.

File: metricbeat/helper/module.go [hunk 4]
```diff
@@ -6,6 +6,7 @@ import (
 	"time"
```
Comment (ruflin) [inline metricbeat/helper/module.go]: No empty line required

File: metricbeat/include/list.go [hunk 5]
```diff
@@ -13,10 +13,13 @@ import (
...
 	"github.com/elastic/beats/metricbeat/helper"
 
-	// List of all metrics to make sure they are registred
+	// List of all metrics to make sure they are registered
 	// Every new metric must be added here
 	_ "github.com/elastic/beats/metricbeat/module/apache/status"
 	_ "github.com/elastic/beats/metricbeat/module/mysql/status"
+
+	// Redis module and metrics
+	_ "github.com/elastic/beats/metricbeat/module/redis"
```
Comment (ruflin) [inline metricbeat/include/list.go]: I would suggest not add add any comments and spaces between the lines here as I plan to automate this generation in the future. But for the moment ok for me.

File: metricbeat/module/redis/info/info.go [hunk 6]
```diff
@@ -129,21 +129,51 @@ func init() {
...
+	redisPools map[string]*rd.Pool
+}
 
-func (m *MetricSeter) Fetch(ms *helper.MetricSet) (events []common.MapStr, err error) {
+// Configure connection pool for each Redis host
+func (m *MetricSeter) Setup(cfg *ucfg.Config) error {
 
-	for _, host := range ms.Config.Hosts {
+	config := struct {
+		Hosts   []string `config:"hosts"`
```
Comment (ruflin) [inline metricbeat/module/redis/info/info.go]: Hosts could be fetched from the module, but ok to parse it again.

**Comment by ruflin** [comment]
@chrsblck Thanks a lot for the contribution. LGTM. Could you squash the commits into one?

I'm thinking of doing some cleanup afterwards. I want to try if I can map ucfg in module somehow, so ucfg does not have be imported every time.

**Comment by chrsblck** [comment]
I'll do that tonight when I get home.

**Comment by ruflin** [comment]
@chrsblck Thanks for the contribution.

**Pull request merged**